metals/src/main/scala/scala/meta/internal/metals/Embedded.scala

```diff
@@ -224,45 +231,48 @@ object Embedded {
 
   private def downloadDependency(
       dep: Dependency,
-      scalaVersion: String,
+      scalaVersion: Option[String],
       classfiers: Seq[String] = Seq.empty,
-      resolution: ResolutionParams = ResolutionParams.create()
-  ): List[Path] =
-    fetchSettings(dep, scalaVersion)
+      resolution: Option[ResolutionParams] = None
+  ): List[Path] = {
+    fetchSettings(dep, scalaVersion, resolution)
       .addClassifiers(classfiers: _*)
-      .withResolutionParams(resolution)
       .fetch()
       .asScala
       .toList
       .map(_.toPath())
+  }
 
   def downloadScalaSources(scalaVersion: String): List[Path] =
     downloadDependency(
       scalaDependency(scalaVersion),
-      scalaVersion,
+      Some(scalaVersion),
       classfiers = Seq("sources")
     )
 
   def downloadScala3Sources(scalaVersion: String): List[Path] =
     downloadDependency(
       scala3Dependency(scalaVersion),
-      scalaVersion,
+      Some(scalaVersion),
       classfiers = Seq("sources")
     )
 
   def downloadSemanticdbScalac(scalaVersion: String): List[Path] =
-    downloadDependency(semanticdbScalacDependency(scalaVersion), scalaVersion)
+    downloadDependency(
+      semanticdbScalacDependency(scalaVersion),
+      Some(scalaVersion)
+    )
   def downloadMtags(scalaVersion: String): List[Path] =
-    downloadDependency(mtagsDependency(scalaVersion), scalaVersion)
+    downloadDependency(mtagsDependency(scalaVersion), Some(scalaVersion))
 
   def downloadMdoc(
       scalaVersion: String,
       scalaBinaryVersion: String,
-      resolutionParams: ResolutionParams = ResolutionParams.create()
+      resolutionParams: Option[ResolutionParams] = None
   ): List[Path] =
     downloadDependency(
       mdocDependency(scalaVersion, scalaBinaryVersion),
-      scalaVersion,
+      scalaVersion = None,
```

Comment:
We don't want to force scala version here since mdoc is released with binary compatibility